cpu/sam0_common/periph/gpio.c

```diff
@@ -217,15 +232,76 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     _EIC->WAKEUP.reg |= (1 << exti);
     /* enable the EIC module*/
     _EIC->CTRL.reg = EIC_CTRL_ENABLE;
-    while (_EIC->STATUS.reg & EIC_STATUS_SYNCBUSY) {}
+    EIC_SYNC();
 #else /* CPU_FAM_SAML21 */
     /* enable the EIC module*/
     _EIC->CTRLA.reg = EIC_CTRLA_ENABLE;
-    while (_EIC->SYNCBUSY.reg & EIC_SYNCBUSY_ENABLE) {}
+    EIC_SYNC();
 #endif
     return 0;
 }
 
+inline static void reenable_eic(gpio_eic_clock_t clock) {
+#if defined(CPU_SAMD21)
+    if (clock == _EIC_CLOCK_SLOW) {
+        GCLK->CLKCTRL.reg = EIC_GCLK_ID
+                          | GCLK_CLKCTRL_CLKEN
+                          | GCLK_CLKCTRL_GEN(SAM0_GCLK_32KHZ);
+    } else {
+        GCLK->CLKCTRL.reg = EIC_GCLK_ID
+                          | GCLK_CLKCTRL_CLKEN
+                          | GCLK_CLKCTRL_GEN(SAM0_GCLK_MAIN);
+    }
+    while (GCLK->STATUS.bit.SYNCBUSY) {}
```

Comment:
A last minute catch: We could use `EIC_SYNC();` here.
Should I change this while squashing?

Reply:
A wait ... I now see this ... it's `GCLK`, not `EIC` ... never mind.

Reply:
Sure, go ahead!